tests/HttpExtractorTest.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\HttpExtractor;
+
+use function file_get_contents;
+use GuzzleHttp\Client;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use GuzzleHttp\Psr7\Uri;
+use PHPUnit\Framework\TestCase;
+use function sys_get_temp_dir;
+use function tmpfile;
+
+class HttpExtractorTest extends TestCase
+{
+    /** @var mixed[] */
+    private $history = [];
+
+    public function testExtract(): void
+    {
+        $resource = new Uri('http://example.com/result.txt');
+        $content = 'File contents';
+        $mockedResponse = new Response(200, [], $content);
+        $client = $this->getMockedGuzzle([$mockedResponse]);
+        $extractor = new HttpExtractor($client);
+        $destination = tempnam(sys_get_temp_dir(), 'http_extractor');
+
+        $extractor->extract($resource, $destination);
+
+        $this->assertSame($content, file_get_contents($destination));
+        $historyItem = array_pop($this->history);
+        /** @var Request $request */
+        $request = $historyItem['request'];
+        $this->assertSame('http://example.com/result.txt', (string) $request->getUri());
```

Comment:
It's better to avoid calling static method 'assertSame' dynamically

Reply:
https://github.com/sebastianbergmann/phpunit/issues/1914#issuecomment-148407321

Reply:
vim, tady https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.static-vs-non-static-usage-of-assertion-methods zase pisou, ze it's a matter of personal preference a muj personal preference je mit tu kontrolu zaplou a nemuset pro ni definovat vyjimky :) just saying

Reply:
ty to odine volas staticky?

Reply:
aha koukam ze jo https://github.com/keboola/input-mapping/blob/683214e65c65e60285909da60880e63be4fc4eed/Tests/ReaderFilesTest.php#L85 
To jsem si nikdy nevsiml:)

Reply:
Za mě optimální se nějak dohodnout a přidat na to check do CS. 

Obecně mám za to, že tyhle věci by se měli jednou dohodnout, pak přidat do CS a pak už je to technikálie, že mi padá build. Ta diskuse by okolo toho měla proběhnout jen jednou. A pak už by se to nikdy nemělo objevit špatně.